File: src/main/java/fi/aalto/cs/apluscourses/utils/parser/FeedbackParser.java
package fi.aalto.cs.apluscourses.utils.parser;

import org.jetbrains.annotations.NotNull;

public class FeedbackParser {

  @NotNull
  public FeedbackParser.TestResults parseTestResults(@NotNull String htmlString) {
    return new TestResults(-1, -1);
  }

  public static class TestResults {
    public final int succeeded;
    public final int failed;

    public TestResults(int succeeded, int failed) {
      this.succeeded = succeeded;
      this.failed = failed;
    }
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/utils/temp/parser/FeedbackParser.kt
package fi.aalto.cs.apluscourses.utils.temp.parser

import fi.aalto.cs.apluscourses.utils.temp.parser.FeedbackParser.TestResults

open class FeedbackParser {
    open fun parseTestResults(htmlString: String): TestResults {
        return TestResults(-1, -1)
    }

    data class TestResults(val succeeded: Int, val failed: Int)
}
